src/services/create-todo.service.ts

```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@nestjs/common';
+import { CacheUpdate } from '@type-cacheable/core';
+import { todoCacheKey } from '@/consts/todo';
+import { Todo } from '@/database/entities/todo.entity';
+import { CreateTodoInput } from '@/inputs/create-todo.input';
+import { TodoRepository } from '@/repositories/todo.repository';
+@Injectable()
+export class CreateTodoService {
+  constructor(private todoRepository: TodoRepository) {}
+  @CacheUpdate({
+    cacheKey: (_, __, todo: Todo) => todo.id,
+    cacheKeysToClear: todoCacheKey,
+  })
+  async create(input: CreateTodoInput): Promise<Todo> {
```

Comment:
What do you think about changing the variable name from "input" to "todo" or "newTodo"?

Reply:
"input" belongs to graphql ecosystem and I think is extra context since you are using "input" when creating something, is very clear.

Reply:
this is a long story
https://dev.to/dcwither/stop-using-data-as-a-variable-name-3954
there is no problem if you use "input", "data", etc if you give meaningful context what is the case, but I just try to avoid as possible,

Reply:
> "input" belongs to graphql ecosystem and I think is extra context since you are using "input" when creating something, is very clear.

I understand your point about what "input" means for you, but I think that "services" should be technology agnostic. It should not matter if you use GraphQL or Rest API.